docs/type-subset.md

```diff
@@ -23,7 +23,7 @@ intersection-td :=
 
 postfix-td := 
   primary-td
-  | postfix-td "[" "]"
+  | postfix-td "[" array-len "]"
```

Comment:
This isn't right with new semantics for dimensions. Adapt what I have done here https://ballerina.io/spec/lang/master/#section_5.4.1.1